commit 93660d1c27b1b85e84621326c0e2c89e00c3fc6f
Author: Peter Eisentraut <peter@eisentraut.org>
Date:   Tue Aug 13 10:01:49 2024 +0200

    Use errmsg_internal for debug messages
    
    Some newer code was applying this inconsistently.

diff --git a/src/backend/postmaster/walsummarizer.c b/src/backend/postmaster/walsummarizer.c
index b62e2c36de9..daa79093821 100644
--- a/src/backend/postmaster/walsummarizer.c
+++ b/src/backend/postmaster/walsummarizer.c
@@ -380,12 +380,12 @@ WalSummarizerMain(char *startup_data, size_t startup_data_len)
 		{
 			List	   *tles = readTimeLineHistory(latest_tli);
 
 			switch_lsn = tliSwitchPoint(current_tli, tles, &switch_tli);
 			ereport(DEBUG1,
-					errmsg("switch point from TLI %u to TLI %u is at %X/%X",
-						   current_tli, switch_tli, LSN_FORMAT_ARGS(switch_lsn)));
+					errmsg_internal("switch point from TLI %u to TLI %u is at %X/%X",
+									current_tli, switch_tli, LSN_FORMAT_ARGS(switch_lsn)));
 		}
 
 		/*
 		 * If we've reached the switch LSN, we can't summarize anything else
 		 * on this timeline. Switch to the next timeline and go around again,
@@ -1215,26 +1215,26 @@ SummarizeWAL(TimeLineID tli, XLogRecPtr start_lsn, bool exact,
 		/* Close temporary file and shut down xlogreader. */
 		FileClose(io.file);
 
 		/* Tell the user what we did. */
 		ereport(DEBUG1,
-				errmsg("summarized WAL on TLI %u from %X/%X to %X/%X",
-					   tli,
-					   LSN_FORMAT_ARGS(summary_start_lsn),
-					   LSN_FORMAT_ARGS(summary_end_lsn)));
+				errmsg_internal("summarized WAL on TLI %u from %X/%X to %X/%X",
+								tli,
+								LSN_FORMAT_ARGS(summary_start_lsn),
+								LSN_FORMAT_ARGS(summary_end_lsn)));
 
 		/* Durably rename the new summary into place. */
 		durable_rename(temp_path, final_path, ERROR);
 	}
 
 	/* If we skipped a non-zero amount of WAL, log a debug message. */
 	if (summary_end_lsn > summary_start_lsn && fast_forward)
 		ereport(DEBUG1,
-				errmsg("skipped summarizing WAL on TLI %u from %X/%X to %X/%X",
-					   tli,
-					   LSN_FORMAT_ARGS(summary_start_lsn),
-					   LSN_FORMAT_ARGS(summary_end_lsn)));
+				errmsg_internal("skipped summarizing WAL on TLI %u from %X/%X to %X/%X",
+								tli,
+								LSN_FORMAT_ARGS(summary_start_lsn),
+								LSN_FORMAT_ARGS(summary_end_lsn)));
 
 	return summary_end_lsn;
 }
 
 /*
@@ -1573,12 +1573,12 @@ summarizer_read_local_xlog_page(XLogReaderState *state,
 					 */
 					private_data->read_upto = switchpoint;
 
 					/* Debugging output. */
 					ereport(DEBUG1,
-							errmsg("timeline %u became historic, can read up to %X/%X",
-								   private_data->tli, LSN_FORMAT_ARGS(private_data->read_upto)));
+							errmsg_internal("timeline %u became historic, can read up to %X/%X",
+											private_data->tli, LSN_FORMAT_ARGS(private_data->read_upto)));
 				}
 
 				/* Go around and try again. */
 			}
 		}
